src/vcpkg/base/downloads.cpp

```diff
@@ -579,6 +582,45 @@ namespace vcpkg
         return res;
     }
 
+    std::string format_url_query(const std::string& base_url, const std::vector<std::string>& query_params)
+    {
+        if (query_params.empty())
+        {
+            return base_url;
+        }
+
+        std::string query = Strings::join("&", query_params);
+
+        return base_url + "?" + query;
+    }
+
+    ExpectedL<std::string> invoke_http_request(const std::string& method,
```

Comment:
Should anything else in `downloads.cpp` now be implemented in terms of this?

Reply:
I don't believe so